### PR TITLE
chore: apply identifier conventions to grants

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
## Changes
- Applied identifier conventions to grant resources
  - Used `helpers.EncodeResourceIdentifier` where possible
  - Used `sdk.ParseXIdentifier` where possible
  - Applied quote ignore on identifier fields
- Added tests to check the migration between the versions

> Note: Some of the changes like `resource_identifier.go`, `custom_diff.go`, `diff_suppressions.go` were taken from #2996

### CODE REVIEW

File: pkg/resources/share.go [hunk 1]
```diff
@@ -164,7 +165,7 @@ func ReadShare(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf("error reading share (%v) err = %w", d.Id(), err)
 	}
-	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
+	if err := d.Set(FullyQualifiedNameAttributeName, share.ID().FullyQualifiedName()); err != nil {
```
Comment (sfc-gh-jmichalak) [inline pkg/resources/share.go:168]: Nit: why? For other resources we use id.FullyQualifiedName() and I think we can be consistent here.

**Review by sfc-gh-jmichalak — CHANGES_REQUESTED** [review]
Overall it looks good. My comments relate to other resources affected by this PR. Also, comments from https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2996 are valid here.

File: pkg/resources/grant_ownership.go [hunk 2]
```diff
@@ -473,26 +478,42 @@ func getOwnershipGrantOn(d *schema.ResourceData) (*sdk.OwnershipGrantOn, error)
...
+func getOwnershipGrantTo(d *schema.ResourceData) (sdk.OwnershipGrantTo, error) {
 	var ownershipGrantTo sdk.OwnershipGrantTo
 
 	if accountRoleName, ok := d.GetOk("account_role_name"); ok {
-		ownershipGrantTo.AccountRoleName = sdk.Pointer(sdk.NewAccountObjectIdentifierFromFullyQualifiedName(accountRoleName.(string)))
+		accountRoleId, err := sdk.ParseAccountObjectIdentifier(accountRoleName.(string))
+		if err != nil {
+			return ownershipGrantTo, err
+		}
+		ownershipGrantTo.AccountRoleName = sdk.Pointer(accountRoleId)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_ownership.go:505]: Yeah, doesn't matter that much but changed.

File: pkg/resources/share.go (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/share.go:168]: Changed

File: pkg/resources/grant_account_role_acceptance_test.go [hunk 3]
```diff
@@ -88,3 +89,94 @@ func TestAcc_GrantAccountRole_user(t *testing.T) {
...
+						Source:            "Snowflake-Labs/snowflake",
+					},
+				},
+				Config: grantAccountRoleBasicConfig(roleName.Name(), parentRoleName.Name()),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_grant_account_role.test", "id", fmt.Sprintf(`%v|ROLE|%v`, roleName.FullyQualifiedName(), parentRoleName.FullyQualifiedName())),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_account_role_acceptance_test.go:116]: Added

File: pkg/resources/grant_application_role_acceptance_test.go [hunk 4]
```diff
@@ -122,3 +123,95 @@ func TestAcc_GrantApplicationRole_application(t *testing.T) {
...
+		},
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"snowflake": {
+						VersionConstraint: "=0.94.1",
+						Source:            "Snowflake-Labs/snowflake",
+					},
+				},
+				Config: grantApplicationRoleBasicConfig(fmt.Sprintf(`\"%s\".\"%s\"`, appRoleId.DatabaseName(), appRoleId.Name()), parentRoleId.Name()),
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_application_role_acceptance_test.go:147]: FQN != raw string of `\"%s\".\"%s\"` (this is how we pass it in terraform config)

File: pkg/resources/grant_ownership_acceptance_test.go [hunk 5]
```diff
@@ -69,11 +69,11 @@ func TestAcc_GrantOwnership_OnObject_Database_ToAccountRole(t *testing.T) {
 
 func TestAcc_GrantOwnership_OnObject_Database_IdentifiersWithDots(t *testing.T) {
 	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifierContaining(".")
-	databaseName := databaseId.Name()
+	databaseName := databaseId.FullyQualifiedName()
 	databaseFullyQualifiedName := databaseId.FullyQualifiedName()
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/grant_ownership_acceptance_test.go:73]: Changed

**Review by sfc-gh-jmichalak — DISMISSED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Pull request merged**